airflow/dags/monitor/monitor.py

```diff
@@ -22,97 +31,267 @@
+    """
+    Aggregates error messages from XCom based on provided keys.
+
+    :param context: The task context provided by Airflow.
+    :param error_keys: A list of keys to pull from XCom.
+    """
+    task_instance = context["task_instance"]
+    error_messages = []
+    for key in error_keys:
+        errors = task_instance.xcom_pull(key=key)
+        if errors:
+            if isinstance(errors, list):
+                error_messages.extend(errors)
+            else:
+                error_messages.append(errors)
+    return error_messages
+
+
+def construct_service_status_message(
+    task_status: list[str], failed_tasks: list[str], error_messages: list[str], external_trigger: bool
+) -> str:
+    """
+    Constructs the status message for Slack based on task statuses and error messages.
+
+    :param task_status: A list of strings representing the successful tasks.
+    :param failed_tasks: A list of strings representing the failed tasks.
+    :param error_messages: A list of error messages from failed task executions.
+    :param external_trigger: A boolean indicating if the DAG was triggered externally.
+    """
+    # Start with reporting failed tasks and errors if there are any
+    service_status_lines = [f":red_circle: {task}" for task in failed_tasks]
+    service_status_lines.extend(error_messages)
+
+    # If the DAG was externally triggered and there are no failures, report all services are up
+    if external_trigger and not failed_tasks:
+        service_status = "\n:large_green_circle: All services are up!\n"
+    else:
+        # Otherwise, combine the statuses and include successful tasks
+        service_status_lines.extend(
+            [f":large_green_circle: {task}" for task in task_status if task not in failed_tasks]
+        )
+        service_status = "\n".join(service_status_lines)
+
+    return service_status
+
+
+@task(trigger_rule=TriggerRule.ALL_DONE)
+def slack_status(**context: Any) -> None:
+    """
+    Sends a status message to Slack. This task will execute even if upstream tasks fail.
+
+    :param context: The context dictionary provided by Airflow.
+    """
+    tis_dagrun = context["ti"].get_dagrun().get_task_instances()
+    task_status, failed_tasks = aggregate_task_statuses(tis_dagrun)
+    error_keys = ["api_errors", "ui_check_error", "weaviate_check_error", "firestore_check_error"]
+    error_messages = aggregate_error_messages(context, error_keys)
+
+    publish_result = bool(failed_tasks) or context["dag_run"].external_trigger
+    service_status = construct_service_status_message(
+        task_status, failed_tasks, error_messages, context["dag_run"].external_trigger
+    )
+
+    # Send the Slack message if needed
+    if publish_result:
         SlackWebhookOperator(
             task_id="slack_alert",
             slack_webhook_conn_id=slack_webhook_conn,
             message=service_status,
+            trigger_rule=TriggerRule.ALL_DONE,
```

Comment:
```suggestion
```
This should not require

Reply:
https://github.com/astronomer/ask-astro/pull/98/commits/8c3a8dc06726416be0c173d82502c4d6fe05c6d8